syntax.php

```diff
@@ -96,14 +80,30 @@ function handle($match, $state, $pos, Doku_Handler $handler) {
      */
     function render($mode, Doku_Renderer $renderer, $data) {
         global $lang;
+        global $INFO;
+        global $ID;
 
         if ($mode == 'xhtml') {
+            // dbglog($data, "renderer init");
+            $id = $ID;
+
+            // If it's a sidebar, get the original id.
+            if ($INFO != null) {
+                $id = $INFO['id'];
+            }
+	    $match = $data[0];
+	    $match = ($match == '.') ? $id : $match;
+            if (strstr($match, ".:")) {
+                resolve_pageid(getNS($id), $match, $exists);
+	    }
+            // dbglog($match, "renderer final");
+
```

Comment:
this section should probably go outside the `if ($mode == 'xhtml') ` block, because we may want to add renderers later eg pdf or odt and this would be be skipped or need to be duplicated.

Reply:
the out-commented `dbglog(...)`  should be removed